File: Moviebase.DAL/Model/Movie.cs
namespace Moviebase.DAL.Model;

public class Movie
{
    public Guid MovieId { get; set; }

    public string Title { get; set; }

    public string Year { get; set; }

    public string PosterId { get; set; }

    public decimal ImdbRating { get; set; }

    public ICollection<MovieGenre> MovieGenres { get; set; }

    public ICollection<MovieActor> MovieActors { get; set; }

    public ICollection<Review>? Reviews { get; set; }
}

File: Moviebase.WebClient/src/app/models/movie.ts
import { Guid } from "typescript-guid";
import { Genre } from "./genre";
import { Actor } from "./actor";

export type Movie = {
    movieId: Guid;
    title: string;
    year: string;
    posterId: string;
    imdbRating: number;
    genres: Genre[];
    actors: Actor[];
}